### PR TITLE
Feature/discord

### PR DESCRIPTION
**Pull request by seunghwako**
# Test before submitting

Before submitting a Pull Request, please test your code. If you created a new created a new feature, then create the test function, too. Please refer to [this document](../CONTRIBUTING.md).

Put the name of the test function of the service you created in the command below.
This is a regular expression that returns all tests containing the character string, so if there are several test codes, you can put a keyword including all of the test codes and turn them.
If you omit the include statement and operate the test, all the test codes we write will work, and myriad errors will occur in your code without environmental variables.

```ts
npm run build && npm run test -- --include TEST_FUNCTION_NAME
```

Of course you won't be able to provide us with your environmental variables, so we'll replace them with our environmental variables to run the program and merge it depending on whether it passes or not.

# Purpose

Please explain to our developers what the features you modified or developed are for.

# Check List

- [ ] Passing an existing or newly created test
- [ ] sufficient summary and description for LLM to understand
- [ ] Does it ensure backward compatibility with existing features, or does the modifications not affect pre-modification request, response type?

### CODE REVIEW

File: src/providers/connector/discord/DiscordProvider.ts [hunk 1]
```diff
@@ -0,0 +1,383 @@
+import { Injectable } from "@nestjs/common";
+import axios from "axios";
+import { IDiscord } from "@wrtn/connector-api/lib/structures/connector/discord/IDiscord";
+import { OAuthSecretProvider } from "../../internal/oauth_secret/OAuthSecretProvider";
+import { IOAuthSecret } from "../../internal/oauth_secret/structures/IOAuthSecret";
+import { ConnectorGlobal } from "../../../ConnectorGlobal";
+
+@Injectable()
+export class DiscordProvider {
+  async getCurrentUser(): Promise<IDiscord.IUser> {
```
Comment (kakasoo) [inline src/providers/connector/discord/DiscordProvider.ts:10]: 실제로는 봇 토큰을 호출해서 봇 정보를 부르는거죠?
함수 이름을 바꾸는 게 나을 거 같습니다.

File: src/providers/connector/discord/DiscordProvider.ts [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+            Authorization: `Bot ${ConnectorGlobal.env.DISCORD_BOT_TOKEN}`,
+          },
+        },
+      );
+      return res.data;
+    } catch (err) {
+      console.error(JSON.stringify(err));
+      throw err;
+    }
+  }
```
Comment (kakasoo) [inline src/providers/connector/discord/DiscordProvider.ts:283]: 히스토리는 토큰이 터질 위험이 있어 페이지네이션 적용 필요.

**Review by kakasoo — COMMENTED** [review]
페이지네이션 추후에라도 꼭 적용해주세요.

File: src/providers/connector/discord/DiscordProvider.ts (hunk 1 above)
Comment (seunghwako) [reply src/providers/connector/discord/DiscordProvider.ts:10]: 함수 삭제 완료

File: src/providers/connector/discord/DiscordProvider.ts (hunk 2 above)
Comment (seunghwako) [reply src/providers/connector/discord/DiscordProvider.ts]: 지금 적용해서 커밋 완료

**Pull request merged**